### ISSUE TITLE
Implement scope and map to user roles/capabilities

### ISSUE THREAD

**Issue by dshanske** [indieweb/wordpress-indieauth#15]

**Comment by dshanske** [comment]
We have scope descriptions. The idea of this is to map the established scopes to a user capability and if they do not have them, then do not grant the scope.

**Comment by dshanske** [comment]
https://developer.wordpress.org/reference/hooks/map_meta_cap/
https://developer.wordpress.org/reference/hooks/user_has_cap/

IndieAuth could use these filters to dynamically filter capabilities.

**Comment by dshanske** [comment]
In pseudo-code:
```
add_filter( 'map_meta_cap', function ( $caps ) {
    foreach ( $caps as $cap ) {
        if ( ! $token->has_cap( $cap ) ) {
            return [ 'do_not_allow' ];
        }
    }
    return $caps;
}, 1000 );
```

**Comment by dshanske** [comment]
Addressed by #169
